File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums) 
        i = 0
        j = n -1
        mid = 0
        while i <= j:
            mid = (i+j)//2
            if target == nums[mid]:
                return mid
            elif target > nums[mid]:
                i = mid+1 
            else:
                j = mid-1
        return -1

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        if(nums.empty()) return;
        int n = nums.size()-1;
        int low = 0;
        int right = n;
        int mid = 0;
        while(mid<=right){
            if(nums[mid]==0) {
                swap(nums[low],nums[mid]);
                low++;
                mid++;
            }
            else if(nums[mid]==1) mid++;
            else{
                swap(nums[mid],nums[right]);
                    right--;
                
            }
    

    }
    }

};